tagpr.go

```diff
@@ -606,28 +606,34 @@ func (tp *tagpr) generatenNextLabels(prIssues []*github.Issue) []string {
 	return nextLabels
 }
 
-func buildChunkSearchIssuesQuery(queryBase string, shasStr string) (chunkQueries []string) {
-	query := queryBase
+func buildChunkSearchIssuesQuery(qualifiers string, shasStr string) (chunkQueries []string) {
+	// array of SHAs
+	keywords := make([]string, 0, 25)
 	// Make bulk requests with multiple SHAs of the maximum possible length.
 	// If multiple SHAs are specified, the issue search API will treat it like an OR search,
-	// and all the pull requests will be searched.u
+	// and all the pull requests will be searched.
 	// This is difficult to read from the current documentation, but that is the current
 	// behavior and GitHub support has responded that this is the spec.
 	for _, sha := range strings.Split(shasStr, "\n") {
 		if strings.TrimSpace(sha) == "" {
 			continue
 		}
 		// Longer than 256 characters are not supported in the query.
+		// Note that the length limit does not include the qualifiers (ex. "repo:owner/repo", "is:close").
 		// ref. https://docs.github.com/en/rest/reference/search#limitations-on-query-length
-		if len(query)+1+len(sha) >= 256 {
-			chunkQueries = append(chunkQueries, query)
-			query = queryBase
+		//
+		// Also, from the results of the experiment, it is possible that when counting
+		// the number of characters in the keyword part, one space character is counted
+		//  as three characters (possibly '%20').
+		if len(strings.Join(keywords, "%20") + "%20" + sha) >= 256 {
```

Comment:
I would have expected something like this

len(keywords) * (3+len(sha)) >= 256 or something close to this.

While I understand the logic of your code, I'm not confident about how this code could be maintained.

Any extra special character added later might break the way to count and may lead to errors.

I would have kept the old code, and check the length of the URL encoded strings

So instead of checking `len(query)+1+len(sha) >= 256`

I would concat a `tmpQuery := query + " " + sha`

Then use `if len(url.QueryEscape(tmpQuery)) >= 256`

If the condition is true, the code is unchanged.

If condition is false, you set

`query = tmpQuery`

Reply:
Thank you for your feedback.

I was actually wondering if I should use the `url` package here.

I will explain my concerns later, but first, let me comment on your suggested fix.

`url.QueryEscape()` escapes spaces to `+` (not `%20`).
To achieve the expected result, we should use `url.PathEscape()` here.

https://go.dev/play/p/0Xk56v95-Ol?v=goprev

However, I wasn't confident that `url.PathEscape()` was the best choice here.
There are two reasons for this:

1. The awkwardness of applying `PathEscape` to **query** escaping.
2. The fact that the GitHub API specification does not disclose how the number of characters in this part is counted (this is determined by black-box testing, but there is no certainty).

Considering these points, do you still think it's better to use the `url` package? I'd like to here your opinion on this.

To be honest, I think it does't matter if we use the `url` package here or not.

Also, I'm a beginner in Go, so I appreciate your feedback like this.

Thanks

Reply:
@ccoVeille 
May I ask your opinion about the above?

Reply:
I didn't receive a notification 🤦‍♂️

I understand it's quite a mess to cope with this.

I thought it was something that was used in URL encoding.

The solution you published today is simpler and safer.

I hope you won't face more trouble.

The changes might come with higher number of request, so maybe the rate limiting will be experienced more often.

Maybe you could/should wait a bit more between calls